pedsnetdcc/prepdb.py

```diff
@@ -0,0 +1,240 @@
+
+
+def _conn_str_with_database(conn_str, dbname):
+    """Adjust a connection string to specify the given database.
+
+    This will override any existing database in the connection string.
+
+    :param conn_str: libpq/psycopg2 connection string
+    :type: str
+    :param dbname: new database name
+    :type: str
+    :return: new connection string
+    :rtype: str
+    """
+    dbname_clause = 'dbname={}'.format(dbname)
+    if 'dbname=' in conn_str:
+        new_conn_str = re.sub(r'dbname=([^ ]+)', dbname_clause, conn_str)
+    else:
+        new_conn_str = conn_str + ' ' + dbname_clause
+    return new_conn_str
+
+
+def _sites_and_dcc(dcc_only=False):
+    """Return a tuple containing "dcc" and the site names, or just "dcc".
+    :param dcc_only: return a tuple containing just 'dcc'
+    :type: bool
+    :rtype: tuple
+    """
+    if dcc_only:
+        return 'dcc',
+    else:
+        return ('dcc', 'chop', 'colorado', 'nationwide', 'nemours', 'seattle',
+                'stlouis')
+
+
+def _version_to_shorthand(version):
+    """Given a version string such as 'X.Y.Z', return 'XY'.
+
+    TODO: This is an awkward convention when X >= 10.
+
+    :param version: PEDSnet version 'X.Y' or 'X.Y.Z'
+    :type: str
+    :rtype: str
+    :raises: ValueError
+    """
+    parts = version.split('.')
+    if len(parts) != 2 and len(parts) != 3:
+        tmpl = 'Version string must be like X.Y or X.Y.Z, not `{}`'
+        raise ValueError(tmpl.format(version))
+    return parts[0] + parts[1]
+
+
+def _make_database_name(model_version):
+    """Return a database name, given a version string, e.g. '21' for '2.1'.
+    :param model_version: PEDSnet model version: X.Y.Z or X.Y
+    :type: str
+    :rtype: str
+    """
+    short_version = _version_to_shorthand(model_version)
+    return 'pedsnet_dcc_v{}'.format(short_version)
+
+
+def _create_database_sql(database_name):
+    """Return a tuple of statements to create the database with the given name.
+    :param database_name: Database name
+    :type: str
+    :rtype: tuple
+    """
+    tmpl = "create database {} with owner = dcc_owner template = template0 " \
+           "encoding = 'UTF8' lc_collate = 'C' lc_ctype = 'C'"
+    return tmpl.format(database_name),
+
+
+def _database_privileges_sql(database_name):
+    """Return a tuple of statements granting privileges on the database.
+    :param database_name: Database name
+    :type: str
+    :return: a tuple of statements
+    :rtype: tuple(str)
+    """
+    return 'grant create on database {} to peds_staff'.format(database_name),
+
+
+# SQL template for creating site schemas in an internal database instance.
+_sql_site_internal_template = """
+create schema if not exists {{.Site}}_pedsnet authorization dcc_owner;
+create schema if not exists {{.Site}}_pcornet authorization dcc_owner;
+create schema if not exists {{.Site}}_harvest authorization dcc_owner;
+create schema if not exists {{.Site}}_achilles authorization dcc_owner;
```

Comment:
Please add `create schema if not exists {{.Site}}_id_maps authorization dcc_owner;` for the id mapping.